pkg/discovery/util/node_util.go

```diff
@@ -185,6 +192,21 @@ func DetectNodeRoles(node *api.Node) sets.String {
 	return allRoles
 }
 
+func DetectNodePools(node *api.Node) sets.String {
+	allPools := sets.NewString()
+	for k, v := range node.Labels {
+		switch {
+		case k == NodePoolAKS && v != "":
```

Comment:
nit: Shall we merge the conditions here with || ?

Reply:
Nit: would it be cleaner to have a private function to check if it is valid node pool label and then simply use a if condition here without switch-case? Like:
```
func DetectNodePools(node *api.Node) sets.String {
	allPools := sets.NewString()
	for k, v := range node.Labels {
		if isValidNodePoolLabel(k) && v != "" {
			allPools.Insert(v)
		}
	}
	return allPools
}

func isValidNodePoolLabel(label string) bool {
	return label == NodePoolAKS || label == NodePoolGKE || strings.Contains(label, NodePoolEKSIdentifier)
}
```

Reply:
Refactored as above